src/Countries/Iran.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Spatie\Holidays\Countries;
+
+use Carbon\CarbonImmutable;
+
+class Iran extends Country
+{
+    public function countryCode(): string
+    {
+        return 'ir';
+    }
+    
+    protected function allHolidays(int $year): array
+    {
+        return array_merge([
+            'Islamic Revolution Day' => '02-11',
+            'Nationalization of Oil Industry Day' => '03-19',
+            'Nowruz First Day' => '03-20',
+            'Nowruz Second Day' => '03-21',
+            'Nowruz Third Day' => '03-22',
+            'Nowruz Fourth Day' => '03-23',
+            'Islamic Republic Day' => '03-31',
+            'Nature Day' => '04-01',
+            'Death of Khomeini Day' => '06-03',
+            'Revolt of Khordad 15 Day' => '06-04',
+        ], $this->variableHolidays($year));
+    }
+    
+    /** @return array<string, CarbonImmutable> */
+    protected function variableHolidays(int $year): array
+    {
+        $easter = CarbonImmutable::createFromTimestamp(easter_date($year))
+            ->setTimezone('Asia/Tehran');
+        
+        return [
+            //add more holidays
```

Comment:
Some countries already solved the Islamic holidays part. 

https://github.com/spatie/holidays/blob/main/src/Countries/Turkey.php#L11

Reply:
![image](https://github.com/spatie/holidays/assets/111766206/a29e234c-d372-4f06-883f-2784b6911b08)
We have **2 types** of holidays in Iran.
In the first step (this PR) I added `national and revolutionary holidays` because these types of holidays are always fixed and **do not change** every year.

We(Iran) **only** share the Ramadan and Eid al-Ghadir holidays with Turkey. So, I'm adding "Religious (Islamic) holidays" to another PR because I need to double-check with the solar calendar.

Reply:
Sounds good. Would it make sense to return Farsi(?) by default instead of English?
You can add translations here: https://github.com/spatie/holidays/tree/main/lang

Reply:
Yes, Thank you for this. Return to Farsi language is much more useful for Farsi language users. ❤️